core/src/main/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateAction.java

```diff
@@ -402,6 +403,18 @@ void validateIncomingState(ClusterState incomingState, ClusterState lastSeenClus
         }
 
         ZenDiscovery.validateStateIsFromCurrentMaster(logger, currentNodes, incomingState);
+        if (lastSeenClusterState != null && lastSeenClusterState.supersedes(incomingState)) {
+            final String message = String.format(
+                    Locale.ROOT,
+                    "received older cluster state version [%s] with uuid [%s] than last seen cluster state [%s] with uuid [%s]",
```

Comment:
can we add that it came from the current master? Also - thinking about this more - this can only happen in our testing right? In practice we only have 1 channel (if someone doesn't mess with some settings). Maybe add a comment about it if so...